packages/flutter/test/cupertino/nav_bar_test.dart

```diff
@@ -1185,6 +1185,103 @@ void main() {
     expect(barItems2.length, greaterThan(0));
     expect(barItems2.any((RichText t) => t.textScaleFactor != 1), isFalse);
   });
+
+  testWidgets(
+    'CupertinoSliverNavigationBar stretches upon over-scroll and bounces back once over-scroll ends', 
+    (WidgetTester tester) async {
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: CupertinoPageScaffold(
+          child: CustomScrollView(
+            controller: ScrollController(),
+            slivers: <Widget>[
+              const CupertinoSliverNavigationBar(
+                trailing: Text('Bar Button'),
+                largeTitle: Text('Large Title'),
+              ),
+              SliverToBoxAdapter(
+                child: Container(
+                  height: 1200.0,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    final Offset initialTrailingTextToLargeTitleOffset = tester.getTopLeft(find.byType(Text).first) - tester.getTopLeft(find.byType(Text).last);
```

Comment:
This is a bit risky since there might be other collisions (such as implicit texts created by other widgets). I'd just use use find.text() to get the exact text content